Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -62,31 +64,33 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
     ) {
         self.apiClient = apiClient
         self.configuration = configuration
-        self.uniqueAssetAPIClient = UniqueAssetAPIClient<CheckoutAttemptIdResponse>(apiClient: apiClient)
+        self.uniqueAssetAPIClient = UniqueAssetAPIClient<InitialAnalyticsResponse>(apiClient: apiClient)
     }
 
     // MARK: - Internal
-    
-    internal func fetchAndCacheCheckoutAttemptIdIfNeeded() {
-        fetchCheckoutAttemptId { _ in /* Do nothing, the point is to trigger the fetching and cache the value  */ }
-    }
 
-    internal func fetchCheckoutAttemptId(completion: @escaping (String?) -> Void) {
+    internal func sendInitialAnalytics(with flavor: AnalyticsFlavor, additionalFields: AdditionalAnalyticsFields?) {
         guard configuration.isEnabled else {
             checkoutAttemptId = "do-not-track"
-            completion(checkoutAttemptId)
             return
         }
+        
+        let analyticsData = AnalyticsData(flavor: flavor,
+                                          additionalFields: additionalFields,
+                                          context: configuration.context)
 
-        let checkoutAttemptIdRequest = CheckoutAttemptIdRequest()
+        fetchCheckoutAttemptId(with: analyticsData)
```

Comment:
It feels a bit confusing that `send` method calls `fetch` in the end, wdyt? Fetching the id seems to be a side-effect here not the primary action

Reply:
Yeah this one is tricky both ways. Can name it `sendData....` then the checkoutAttemptId would be hidden. `sendDataAndFetchCheckoutAttemptId...` :) is the most expressive but it's long

Reply:
Also now the function is really short, we can maybe remove the 2nd function and keep sending the request here as well, wdyt?

Reply:
Yes, seems new changes made everything more complex!

Reply:
This turned out nice! 63016fdaebcce42e664feb3212e1552a274e8723

Reply:
Neat! 👏